core/src/test/java/com/microsoft/applicationinsights/internal/config/connection/ConnectionStringParsingTests.java

```diff
@@ -0,0 +1,254 @@
+        final String cs2 = "SnapshotEndpoint="+ snapshot+";ProfilerEndpoint="+ profiler+";InstrumentationKey="+ ikey +";LiveEndpoint="+ live;
+
+        TelemetryConfiguration config2 = new TelemetryConfiguration();
+
+        parseInto(cs1, config);
+        parseInto(cs2, config2);
+
+        assertEquals(config.getInstrumentationKey(), config2.getInstrumentationKey());
+        assertEquals(config.getEndpointConfiguration().getIngestionEndpoint(), config2.getEndpointConfiguration().getIngestionEndpoint());
+        assertEquals(config.getEndpointConfiguration().getLiveEndpoint(), config2.getEndpointConfiguration().getLiveEndpoint());
+        assertEquals(config.getEndpointConfiguration().getProfilerEndpoint(), config2.getEndpointConfiguration().getProfilerEndpoint());
+        assertEquals(config.getEndpointConfiguration().getSnapshotEndpoint(), config2.getEndpointConfiguration().getSnapshotEndpoint());
+    }
+
+    @Test
+    public void endpointWithNoSchemeIsHttps() throws ConnectionStringParseException {
+        parseInto("InstrumentationKey=fake-ikey;IngestionEndpoint=my-ai.example.com", config);
+        assertEquals("https", config.getEndpointConfiguration().getIngestionEndpoint().getScheme());
+    }
+
+    @Test
+    public void httpEndpointKeepsScheme() throws ConnectionStringParseException {
+        parseInto("InstrumentationKey=fake-ikey;IngestionEndpoint=http://my-ai.example.com", config);
+        assertEquals("http", config.getEndpointConfiguration().getIngestionEndpoint().getScheme());
+    }
+
+    @Test
+    public void emptyIkeyValueIsInvalid() throws ConnectionStringParseException {
+        exception.expect(InvalidConnectionStringException.class);
+        final String cs = "InstrumentationKey=;IngestionEndpoint=https://ingestion.example.com;EndpointSuffix=ai.example.com";
+        parseInto(cs, config);
+    }
+
+    @Test
+    public void multipleKeySeparatorsIsInvalid() throws ConnectionStringParseException {
+        exception.expect(InvalidConnectionStringException.class);
+        final String ikey = "fake-ikey";
+        final String cs = "Authorization=ikey;InstrumentationKey=="+ikey;
+        parseInto_printExceptionAndRethrow(cs);
+    }
+
+    @Test
+    public void emptyStringIsInvalid() throws ConnectionStringParseException {
+        exception.expect(InvalidConnectionStringException.class);
+        parseInto("", config);
+    }
+
+    @Test
+    public void nonKeyValueStringIsInvalid() throws ConnectionStringParseException {
+        exception.expect(InvalidConnectionStringException.class);
+        parseInto(UUID.randomUUID().toString(), config);
+    }
+
+    @Test
+    public void missingAuthorizationIsInvalid() throws ConnectionStringParseException {
+        exception.expect(InvalidConnectionStringException.class);
+        parseInto("LiveEndpoint=https://live.example.com", config);
+    }
```

Comment:
nit: Spec states that Authorization is optional because we currently only support the "ikey" value.
You're technically not wrong to require it so not a deal-breaker.